### PR TITLE
Published Workflow Sharing Page Overhaul

### LINKED ISSUE #16379: public shareable workflow page needs some :heart:
The following page is what people see if they got a link to a public shared workflow.

grafik

We can do much better on this page :)

* make the "import workflow" and "download workflow" buttons bigger and more central, I assume that is the most important action on this site
* include more workflow information metadata (annotation, license, tags)
* include versions?
* A read-only workflow view would be nice to include
* A workflow run button/function

### PR DESCRIPTION
**Pull request by ElectronicBlueberry**
closes #16379

Overhaul of the Workflow Sharing page to be more informative, visually appealing, and easier to navigate.

image

For comparison, this is what the page looked like before the overhaul:

image

The changes break down as follows:

* added a new read-only mode to the workflow editor
* made workflow editors css self-contained to ease embedding
* embedded interactive read-only view of the workflow editor in the sharing page
* moved workflow information out of the side panel, to be more visible
* made page responsive, to be usable on smaller screens
* added more information about workflow, to mirror what can be set in the editor
* added run button
* made import and download buttons more obvious
* add activity bar to published items
* refactored NodeInput and NodeOutput

I would like some feedback from somebody more familiar with the back-end for workflows about the following change:

* changed load_workflow endpoint to be accessible for shared workflows

I could not see any reason why this endpoint should only be callable for the workflows owner, but may have missed something. Please let me know if I did.

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

**Comment by mvdbeek** [comment]
This is absolutely amazing, it looks very clean, thanks so much!

File: client/src/components/Workflow/Published/WorkflowPublished.vue [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+
+const loading = ref(true);
+const errored = ref(false);
+const errorMessage = ref("");
+
+const { datatypesMapper } = useDatatypesMapper();
+
+const connectionsStore = useConnectionStore();
+const stepStore = useWorkflowStepStore();
+const stateStore = useWorkflowStateStore();
```
Comment (mvdbeek) [inline client/src/components/Workflow/Published/WorkflowPublished.vue]: Do we need to scope these stores somehow, e.g. to the workflow id ? It's definitely not a blocker, but if we start using them throughout more parts of the app I guess we need to be careful ?

Comment (ElectronicBlueberry) [reply client/src/components/Workflow/Published/WorkflowPublished.vue]: Yes, I think scoping these to the workflow id is the way to go. I'd like to handle that in a separate PR however.

**Comment by bgruening** [comment]
The failing selenium tests, especially the tours, look relevant. Thanks @ElectronicBlueberry

**Review by mvdbeek — APPROVED** [review]
This is so cool, and it works really well. Thanks a lot @ElectronicBlueberry! Can't wait to see more applications of this :)

**Pull request merged**